code/ship/ship.cpp

```diff
@@ -10334,6 +10334,8 @@ int ship_fire_primary(object * obj, int stream_weapons, int force)
 				if (winfo_p->wi_flags3 & WIF3_APPLY_RECOIL){
 					firepoint_list = new vec3d[numtimes * points];
 					vm_vec_zero(&total_impulse);
+				} else {
+					firepoint_list = nullptr;
```

Comment:
Kind of hard for me to tell for sure.  Here's a log excerpt:
```
CompileC build/FS2_Open.build/Debug/FS2_Open.build/Objects-normal/i386/plr_convert.o /Users/cliff.gordon/fs2_open_git/code/pilotfile/plr_convert.cpp normal i386 c++ com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/cliff.gordon/fs2_open_git/projects/Xcode4
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -arch i386 -fmessage-length=218 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -fcolor-diagnostics -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -D_DEBUG -DSCP_UNIX -DUSE_OPENAL -DAPPLE_APP -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -mmacosx-version-min=10.6 -g -fvisibility=hidden -fvisibility-inlines-hidden -Wno-sign-conversion -I/Users/cliff.gordon/fs2_open_git/projects/Xcode4/build/FS2_Open.build/Debug/FS2_Open.build/FS2_Open\ 3.7.3\ (debug).hmap -I/Users/cliff.gordon/fs2_open_git/projects/Xcode4/build/Debug/include -I../../code -I../../libjpeg -I../../lua -IFrameworks/SDL.framework/Headers -I/Users/cliff.gordon/fs2_open_git/projects/Xcode4/build/FS2_Open.build/Debug/FS2_Open.build/DerivedSources/i386 -I/Users/cliff.gordon/fs2_open_git/projects/Xcode4/build/FS2_Open.build/Debug/FS2_Open.build/DerivedSources -Wno-write-strings -Wno-char-subscripts -Wno-shorten-64-to-32 -F/Users/cliff.gordon/fs2_open_git/projects/Xcode4/build/Debug -FFrameworks -MMD -MT dependencies -MF /Users/cliff.gordon/fs2_open_git/projects/Xcode4/build/FS2_Open.build/Debug/FS2_Open.build/Objects-normal/i386/plr_convert.d --serialize-diagnostics /Users/cliff.gordon/fs2_open_git/projects/Xcode4/build/FS2_Open.build/Debug/FS2_Open.build/Objects-normal/i386/plr_convert.dia -c /Users/cliff.gordon/fs2_open_git/code/pilotfile/plr_convert.cpp -o /Users/cliff.gordon/fs2_open_git/projects/Xcode4/build/FS2_Open.build/Debug/FS2_Open.build/Objects-normal/i386/plr_convert.o
```
I'm not sure what the '1.0' is at the beginning of that, but when I run the clang path with --version I get:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang --version
Apple LLVM version 5.1 (clang-503.0.40) (based on LLVM 3.4svn)
Target: x86_64-apple-darwin12.5.0
Thread model: posix